community/php7/APKBUILD

```diff
@@ -1,152 +1,317 @@
+	ctype
+	curl
+	dba
+	dom
+	enchant
+	exif
+	fileinfo
+	ftp
+	gd
+	gettext
+	gmp
+	iconv
+	imap
+	intl
+	json
+	ldap
+	mbstring
+	mcrypt
+	mysqli
+	mysqlnd
+	odbc
+	opcache
+	openssl
+	pcntl
+	pdo
+	pdo_dblib
+	pdo_mysql
+	pdo_odbc
+	pdo_pgsql
+	pdo_sqlite
+	pgsql
+	phar
+	posix
+	pspell
+	recode
+	session
+	shmop
+	simplexml
+	snmp
+	soap
+	sockets
+	sqlite3
+	sysvmsg
+	sysvsem
+	sysvshm
+	tidy
+	tokenizer
+	wddx
+	xml
+	xmlreader
+	xmlrpc
+	xmlwriter
+	xsl
+	zip
+	zlib
 	"
-subpackages="$pkgname-dev $pkgname-doc $pkgname-apache2 $pkgname-phpdbg $pkgname-embed
-	$pkgname-litespeed $pkgname-cgi $pkgname-fpm $pkgname-pear::noarch
-	"
-for _ext in $_exts; do
+for _ext in $_extensions; do
 	case "$_ext" in
-	*:*) subpackages="$subpackages $pkgname-$_ext";;
-	*) subpackages="$subpackages $pkgname-$_ext:_package_ext";;
+		phar) subpackages="$subpackages $pkgname-$_ext:$_ext";;
+		*) subpackages="$subpackages $pkgname-$_ext:_extension";;
 	esac
 done
 subpackages="$subpackages $pkgname-common::noarch"
 
-_apiver="20151012"
 
 prepare() {
 	cd "$builddir"
 
-	default_prepare || return 1
-	update_config_sub || return 1
+	default_prepare
+	update_config_sub
 
 	local vapi=$(sed -n '/#define PHP_API_VERSION/{s/.* //;p}' main/php.h)
 	if [ "$vapi" != "$_apiver" ]; then
 		error "Upstreram API version is now $vapi. Expecting $_apiver"
+		error "After updating _apiver, all 3rd-party extensions must be rebuilt."
 		return 1
 	fi
+
+	# https://bugs.php.net/63362 - Not needed but installed headers.
+	# Drop some Windows specific headers to avoid installation,
+	# before build to ensure they are really not needed.
+	rm -f TSRM/tsrm_win32.h \
+		TSRM/tsrm_config.w32.h \
+		Zend/zend_config.w32.h \
+		ext/mysqlnd/config-win.h \
+		ext/standard/winver.h \
+		main/win32_internal_function_disabled.h \
+		main/win95nt.h
+
+	# Fix some bogus permissions.
+	find . -name \*.[ch] -exec chmod 644 {} \;
+
+	# XXX: Delete failing tests.
+	sed -n '/^[^#]/p' "$srcdir"/disabled-tests.list | while read item; do
+		rm -r $item  # do it in this way to apply globbing...
+	done
+
 	autoconf
 }
 
+# Notes:
+# * gd-jis-conv breaks any non-latin font rendering (vakartel).
+# * libxml cannot be build as shared.
+# * Doesn't work with system-provided onigurama, some tests fail (invalid code
+#   point); probably because bundled onigurama is version 5.x, but we have 6.x.
 _build() {
-	[ "$CARCH" = "s390x" ] && _disable_pcre_jit="--without-pcre-jit"
-	export EXTENSION_DIR=/usr/lib/$pkgname/modules
-	./configure \
+	local without_pcre_jit
+	[ "$CARCH" = "s390x" ] && without_pcre_jit="--without-pcre-jit"
+
+	EXTENSION_DIR=$_extension_dir ./configure \
 		--build=$CBUILD \
 		--host=$CHOST \
 		--prefix=/usr \
-		--program-suffix=7 \
-		--libdir=/usr/lib/$pkgname \
+		--program-suffix=$_suffix \
+		--libdir=$_libdir \
 		--datadir=/usr/share/$pkgname \
 		--sysconfdir=/etc/$pkgname \
 		--localstatedir=/var \
 		--with-layout=GNU \
 		--with-pic \
 		--with-pear=/usr/share/$pkgname \
```

Comment:
There is no sence to store source-based extensions in versionized subdirectory, because it fits any php version.

This how it done in my variant: --with-pear=/usr/share/pear

And also all php[57]-pear-* packages must be remade to be unversionized.